test/unit/calculators/benefit_cap_calculator_configuration_test.rb

```diff
@@ -50,7 +50,7 @@ class BenefitCapCalculatorConfigurationTest < ActiveSupport::TestCase
 
       context "Flow configuration" do
         setup do
-          BenefitCapCalculatorConfiguration.any_instance.stubs(:data).returns(
+          BenefitCapCalculatorConfiguration.stubs(:data).returns(
```

Comment:
And this is why not using a module would be a better pattern!

`BenefitCapCalculatorConfiguration` should be a class with `data` passed into it on initialization. That way the tests become a lot simpler. For example, if you want to test the `benefits` method:

```ruby
benefits = {foo: :bar}
data = {benefits: benefits}
config = BenefitCapCalculatorConfiguration.new(data)
assert_equal benefits, config.benefits
```
That means each method can be tested cleanly without stubbing and without having to load unnecessary data. Because the test data is simpler, the tests become useful documentation on exactly what each method is doing.

To tidy it up, you could add a class method like this:
```ruby
def self.with_yaml_file(path)
  data = YAML.load_file(path).with_indifferent_access
  new(data)
end
```
Then the location of the config file is defined within the flow using it, rather than having to dig into the configuration object to find it:
```ruby
config = BenefitCapCalculatorConfiguration.with_yaml_file(Rails.root.join("config/smart_answers/benefit_cap_data.yml"))
```

Reply:
Yup - I can see what you are saying. Considering this was the existing behaviour, is not making these changes are a blocker for this PR?